packages/calling-component-bindings/src/providers/CallProvider.tsx

```diff
@@ -57,5 +60,26 @@ export const CallProvider = (props: CallProviderProps): JSX.Element => <CallProv
  * @public
  */
 export const useCall = (): Call | undefined => {
-  return useContext(CallContext)?.call;
+  const call = useContext(CallContext)?.call;
+  if (call && !isACSCall(call)) {
+    throw new Error('TeamsCall object was provided, try useTeamsCall() instead');
+  }
+  return call;
+};
+
+/* @conditional-compile-remove(teams-identity-support) */
+/**
+ * Hook to obtain {@link @azure/communication-calling#TeamsCall} from the provider.
+ *
+ * Useful when implementing a custom component that utilizes the providers
+ * exported from this library.
+ *
+ * @public
+ */
+export const useTeamsCall = (): undefined | /* @conditional-compile-remove(teams-identity-support) */ TeamsCall => {
```

Comment:
This should also be marked `@beta` not `@public`.

Also consider simply conditionally compiling out the entire method (simpler than conditionally compiling the argument and then also the export in `index.ts`. Your call ;)